handler.go

```diff
@@ -17,10 +17,12 @@ import (
 var errResourceNotFound = errors.New("resource not found")
 
 func notFoundError(w http.ResponseWriter, err error) {
+	cacheHeaders(w, 0)
```

Comment:
I think you need to change the internal implementation. The recommended for Cache-Control value for not cachable resources is:
`Cache-Control: no-cache, no-store, must-revalidate`

see: https://developer.mozilla.org/pl/docs/Web/HTTP/Headers/Cache-Control (Preventing caching)

Reply:
WDYT about using "private"? https://docs.fastly.com/en/guides/configuring-caching#do-not-cache

Reply:
Hm.. "private" is to inform the intermittent servers not to cache data, because they will be cached by browsers. To prevent caching on the browser level, AFAIK you need to add `Max-Age: 0`.

Reply:
Referring to the Mozilla docs, `Cache-Control: no-store` should be sufficient at this point:


>no-store
>The response may not be stored in any cache. Although other directives may be set, this alone is the only directive you need in preventing cached responses on modern browsers. max-age=0 is already implied. Setting must-revalidate does not make sense because in order to go through revalidation you need the response to be stored in a cache, which no-store prevents.

Reply:
Fastly does not respect `no-store`:

```
Fastly does not currently respect no-store or no-cache directives. Including either or both of these in a Cache-Control header has no effect on Fastly's caching decision, unless you alter this behavior using custom VCL.
```